### PR TITLE
Decoupled the upsert command and destroy command

### PR DESCRIPTION
**Pull request by tishen25**
- Separated upsert command and destroy command into different `null_resources`.
- Pinned Terraform version in makefile

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_terraform
 Failed Terraform check. More details below. 
 ```bash 
Running terraform fmt
main.tf
--- old/main.tf
+++ new/main.tf
@@ -220,11 +220,11 @@
   ]
 
   triggers = merge({
-    md5                    = md5(var.create_cmd_entrypoint)
-    arguments              = md5(var.create_cmd_body)
-    create_cmd_entrypoint  = var.create_cmd_entrypoint
-    create_cmd_body        = var.create_cmd_body
-    gcloud_bin_abs_path    = local.gcloud_bin_abs_path
+    md5                   = md5(var.create_cmd_entrypoint)
+    arguments             = md5(var.create_cmd_body)
+    create_cmd_entrypoint = var.create_cmd_entrypoint
+    create_cmd_body       = var.create_cmd_body
+    gcloud_bin_abs_path   = local.gcloud_bin_abs_path
   }, var.create_cmd_triggers)
 
   provisioner "local-exec" {
Error: terraform fmt failed with exit code 3
Check the output for diffs and correct using terraform fmt <dir>
 ```

File: main.tf [hunk 1]
```diff
@@ -208,7 +208,7 @@ resource "null_resource" "gcloud_auth_google_credentials" {
   }
 }
 
-resource "null_resource" "run_command" {
+resource "null_resource" "run_upsert_command" {
```
Comment (morgante) [inline main.tf:211]: We shouldn't rename the command (as it will cause an unnecessary state migration).

File: main.tf [hunk 2]
```diff
@@ -237,6 +235,25 @@ resource "null_resource" "run_command" {
     EOT
   }
 
+}
+
+resource "null_resource" "run_destroy_command" {
```
Comment (morgante) [inline main.tf:240]: What happens with this the first time you run? Anything?

Comment (tishen25) [reply main.tf:240]: The resource gets created. It shows up in the plan like this:
```
# module.agent-policy.module.gcloud.null_resource.run_destroy_command[0] will be created
  + resource "null_resource" "run_destroy_command" {
      + id       = (known after apply)
      + triggers = {
          + "destroy_cmd_body"       = "<my-project-id> \"<my-policy-id>\""
          + "destroy_cmd_entrypoint" = "agent-policy/scripts/delete-script.sh"
          + "gcloud_bin_abs_path"    = "/google-cloud-sdk/bin"
        }
    }
```

File: main.tf (hunk 1 above)
Comment (tishen25) [reply main.tf:211]: Makes sense!

File: main.tf (hunk 2 above)
Comment (morgante) [reply main.tf:240]: Ok great, sounds like it does exactly what we want. 👍

Comment (tishen25) [reply main.tf:240]: Nice!

**Pull request merged**